flake8_trio.py

```diff
@@ -1421,9 +1448,89 @@ def visit_AsyncFunctionDef(
 
     def visit_Call(self, node: ast.Call):
         if self.async_function and not getattr(node, "awaited", False):
-            blocking_calls = self.options.trio200_blocking_calls
-            if key := fnmatch_qualified_name([node.func], *blocking_calls):
-                self.error(node, key, blocking_calls[key])
+            self.visit_blocking_call(node)
+
+    def visit_blocking_call(self, node: ast.Call):
+        blocking_calls = self.options.trio200_blocking_calls
+        if key := fnmatch_qualified_name([node.func], *blocking_calls):
+            self.error(node, key, blocking_calls[key])
+
+
+@error_class
+class Visitor21X(Visitor200):
+    error_codes = {
+        "TRIO210": "Sync HTTP call {} in async function should use httpx.AsyncClient",
+        "TRIO211": "Sync HTTP call {} in async function should use httpx.AsyncClient",
+        "TRIO212": "Sync HTTP call {} in async function should use httpx.AsyncClient",
```

Comment:
I think if we're giving the same error message, that's a signal that it should be a single error code.

Imprecise heuristics are something of an exception since they should be separately disablable... but not really because the message should also make it clear we're guessing.  (see below)

Reply:
yeah it was more 22X that got me going on separate error codes, so I figured I'd make them separate for the moment. But esp when it's only the originating library that differs, and users aren't really gonna use too many different ones and care about the distinction within async code, these can probably all be collapsed.

Should the code be 210 or 201 (moving 22x to 21x)?

Reply:
Let's go with 21x: I think it's useful to have the middle digit group like warnings (e.g. "http requests"), and the last to distinguish specific messages/responses/confidence levels/etc.